app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/NamespaceSelector.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import Select, { Creatable, Option } from 'react-select';
+import { get, find, map, includes } from 'lodash';
+import { IAccountDetails } from '@spinnaker/core';
+
+export interface INamespaceSelectorProps {
+  onChange: (namespace: string) => void;
+  accounts: IAccountDetails[];
+  selectedAccount: string;
+  selectedNamespace: string;
+  createable?: boolean;
+}
+
+export class NamespaceSelector extends React.Component<INamespaceSelectorProps> {
+  public defaultProps = { createable: false };
+
+  private getNamespaceOptions(): Array<Option<string>> {
+    const { accounts, selectedAccount } = this.props;
+    const selectedAccountDetails = find(accounts, a => a.name === selectedAccount);
+    const namespaces = get(selectedAccountDetails, 'namespaces', []);
+    const options = map(namespaces, n => ({ label: n, value: n }));
+    if (this.props.createable && !includes(namespaces, this.props.selectedNamespace)) {
```

Comment:
In the situation where a user has selected a namespace available in a particular account, and then changes the account to one where that namespace is no longer an option, we will keep that namespace available as an option, which might be confusing. Wonder if there's any way to designate a particular option as being user-created and only persist those between account re-selects?

Reply:
Oh, that's a good catch. I didn't consider that. What if we just limited it to options containing `${`? That should satisfy the use case without having to worry about the above. The option would only persist if you're using SPeL.

Reply:
love it, ship it